dio/test/interceptor_test.dart

```diff
@@ -469,6 +469,40 @@ void main() {
     });
   });
 
+  group('Error Interceptor', () {
+    test('handled when request cancelled', () async {
+      final cancelToken = CancelToken();
+      DioError? iError, qError;
+      final dio = Dio()
+        ..httpClientAdapter = MockAdapter()
+        ..options.baseUrl = MockAdapter.mockBase
+        ..interceptors.add(
+          InterceptorsWrapper(
+            onError: (DioError e, ErrorInterceptorHandler handler) {
+              iError = e;
+              handler.next(e);
+            },
```

Comment:
Actually, the caller will be called previously, but it'll be slightly after the expectation. If we add any delay after awaiting the request, the test will be passed.